openfermioncirq/utils/qubitoperator_translator.py

```diff
@@ -0,0 +1,72 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import cirq
+from openfermion import QubitOperator
+
+rotation_dict = {'X': cirq.X, 'Y': cirq.Y, 'Z': cirq.Z}
+
+
+def _qubitoperator_to_pauli_string(qubit_op: QubitOperator) -> cirq.PauliString:
+    """
+    Convert QubitOperator to Pauli String.
+
+    Args:
+        qubit_op (QubitOperator): operator to convert.
+
+    Returns:
+        pauli_string (PauliString): cirq PauliString object.
+
+    Raises:
+        TypeError: if qubit_op is not a QubitOpertor.
+        ValueError: if qubit_op has more than one Pauli string.
+    """
+    if not isinstance(qubit_op, QubitOperator):
+        raise TypeError('Input must be a QubitOperator.')
+    if len(qubit_op.terms) > 1:
+        raise ValueError('Input has more than one Pauli string.')
+
+    pauli_string = cirq.PauliString()
+    ind_ops, coeff = next(iter(qubit_op.terms.items()))
+
+    if ind_ops == ():
+        return pauli_string * coeff
+
+    else:
+        for ind, op in ind_ops:
+
+            pauli_string *= rotation_dict[op](cirq.LineQubit(ind))
+
+    return pauli_string * coeff
+
+
+def qubitoperator_to_pauli_sum(qubit_op: QubitOperator) -> cirq.PauliSum:
```

Comment:
I'm trying to find the best way to solve this comment. I agree that allowing a qubit list gives much more flexibility.

However, if given a list of qubits, the index of the QubitOperator corresponds to the position of the qubit in the list, or ones needs a map between the indeces and the qubits.

> This is also similar to how QubitOperator/FermionOperator doesn't currently initialise the actual qubits for us.
I don't really understand what you mean by the above sentence.

Reply:
> However, if given a list of qubits, the index of the QubitOperator corresponds to the position of the qubit in the list, or ones needs a map between the indeces and the qubits.

Position of the qubit in list corresponds to the index of the QubitOperator should be enough here since different qubit types often have an implicit ordering defined. This also aligns well with rest of the design, for eg: `DensePauliString` to `PauliString` conversion in Cirq follows a similar approach ([link](https://github.com/quantumlib/Cirq/blob/2bc997325fc3166f987dd6b97e2bdb20a0b89c28/cirq/ops/dense_pauli_string.py#L272))

I like the modified version of this function. :)